packages/open-payments/src/client/requests.ts

```diff
@@ -158,23 +158,34 @@ const handleError = async (
   let errorDescription
   let errorStatus
   let validationErrors
+  let errorCode
 
   const { HTTPError } = await import('ky')
 
   if (error instanceof HTTPError) {
-    let responseBody
+    let responseBody:
+      | {
+          error: { description: string; code: string }
+        }
+      | string
+      | undefined
 
     try {
-      responseBody = (await error.response.json()) as { message: string }
+      responseBody = await error.response.text()
```

Comment:
once we start returning object response from the RS, this can be just `await error.response.json()`, but this handles both